src/query/storage/promremote/query_coverter.go

```diff
@@ -61,14 +63,19 @@ func convertWriteQuery(queries []*storage.WriteQuery) *prompb.WriteRequest {
 				Value: string(tag.Value),
 			})
 		}
-
+		sampleCount += len(query.Datapoints())
 		samples := make([]prompb.Sample, 0, len(query.Datapoints()))
 		for _, dp := range query.Datapoints() {
 			samples = append(samples, prompb.Sample{
 				Value:     dp.Value,
 				Timestamp: dp.Timestamp.ToNormalizedTime(time.Millisecond),
 			})
 		}
+		// Need to make sure the samples meet remote write spec:
+		// https://prometheus.io/docs/concepts/remote_write_spec/#ordering
+		sort.Slice(samples, func(i, j int) bool {
+			return samples[i].Timestamp < samples[j].Timestamp
+		})
```

Comment:
nice finding. What's the consequence if the samples are not sorted?

Reply:
it is a 429 error which might be rejected and hidden, I am working on fixing 429 in thanos to only remove conflicting timestamp not others